src/ode_solver/ode_solver_base.cpp

```diff
@@ -148,10 +151,10 @@ int ODESolverBase<dim,real,MeshType>::steady_state ()
                 this->dg->output_results_vtk(file_number);
             }
         }
-
-        if ( refine && this->residual_norm < 1e-9 && i_refine < n_refine) {
-            i_refine++;
-            dg->refine_residual_based();
+        
+        if (this->residual_norm < meshadaptation->critical_residual)
```

Comment:
The refine_mesh flag seems to only get initialized in the initialize_steady_polynomial_ramping function. What if the steady_state function is called directly?

Reply:
In the new commit, refine_mesh_in_ode_solver is set `true` to begin with.

If someone calls steady_state() directly and does not want mesh adaptation, the default parameter values of total_refinement_cycles and current_refinement_cycle are set to 0 and adapt_mesh() will not be called.

If someone wants to use steady_state() with mesh_adaptation, they can modify the values of total_refinement_cycles in the parameter file.